rfc/ipni-w3c.md

```diff
@@ -146,67 +145,93 @@ flowchart TD
 
 The ipni/offer is optional, and if not invoked, then the user's data is not indexed.
 
-If opting-in to IPNI indexing, the CAR file and the CARx-index are created by the client. The client may choose which multihashes they want indexed and specify this in the ipni/offer request.
+If opting-in to IPNI indexing, a request in to form of an invocation UCAN is created by the client and refers to the content claims bundle that was created when the client stored a data blob. The client may choose which multihashes they want indexed and specify this in the request. By default all multihashes in the inclusion claim are indexed if opting to index data.
 
 ### Indexing stored data
 
-```mermaid
+``mermaid
 sequenceDiagram
     actor Alice
-    Alice->>w3s: put CAR + CARx-index
+    Alice->>w3s: put blob + inclusion claim
     activate w3s
     w3s-->>w3s: create location claims
-    w3s-->>w3s: create/store claims bucket
-    w3s-->>Alice: location claims
+    w3s-->>w3s: create/store claims bundle
+    w3s-->>Alice: claims bundle ID and URL
     deactivate w3s
-    activate Alice
-    Alice->>w3s: ipni/offer (inclucion, locations CIDs)
-    deactivate Alice
+    Alice->>w3s: get blob status
+    activate w3s
+    w3s-->>Alice: ok, claims bundle CID
+    deactivate w3s
+    Alice->>w3s: ipni/offer (claims-CID, opt: mh set))
     activate w3s
-    w3s-->>w3s: fetch & verify CARx-index
+    w3s-->>w3s: fetch & verify claims bundle
     w3s-->>w3s: write advertisement
-    w3s-)ipni: publish head (CID)
+    w3s-)ipni: announce head (CID)
     deactivate w3s
     ipni-->>w3s: fetch advert
     activate ipni
     ipni-->>ipni: index entries
     deactivate ipni
 ```
+- User creates their data blob and inclusion claim, then uploads these to w3s to the pre-signed PUT URL.
+  - The inclusion claim listing sub-blocks within the blob is created by the client whether or not they intend to have their data indexed. This is because these blocks may still be needed to find data within the blob, and to allow indexing to be done at a later point without having the re-read the entire blob.
+  - W3s does not need to immediately verify the correctness of the inclusion claim, but can at any time and then replace the inclusion claim with its own to certify that it was verified.
+- w3s creates the location claim for the blob after verifying the content was uploaded. Multiple location claims are created if the data is partitioned into multiple blobs or if the blob is stored in multiple locations.
+  - The client triggers w3s to create the blob location claim(s) when the client is done uploading the data blob(s) to the URLs issued by w3s. By creating the location claims, w3s is responsible for the location of the data, and should verify the data is present before creating the claims.
```

Comment:
I thought we gave the user a URL to PUT the data at. If that is the case, then how will we know when the user has uploaded their data to that location? This is why I specified that the user will tell w3s.

It seems like what you are saying is that the user is going to upload the data to w3s. This means that w3s must then upload that to the storage location. That would incur a double ingress cost.

Need to make a decision here.

Reply:
You are correct on invocation client gets URL where it uploads the content. I’m just trying to avoid specifying details on how we discover that upload was complete. And trying to highlight that when they’re done uploading they’ll get commitment a.k.a location claim from us.

perhaps I didn’t phrased it very clearly,  feel free to change it.

Reply:
All good. I used a slight editing of your words.